qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -139,79 +133,136 @@ def __init__(
-                    include_custom=self._include_custom,
-                )
-            )
+        # TODO remove this once the stateful methods are deleted
+        self._ret = None
 
-    def _set_expectation(self, exp: ExpectationBase) -> None:
-        self._expectation = exp
-        self._user_valid_expectation = False
-        self._expect_op = None
+    @property
+    def ansatz(self) -> Optional[QuantumCircuit]:
+        """Returns the ansatz."""
+        return self._ansatz
+
+    @ansatz.setter
+    def ansatz(self, ansatz: Optional[QuantumCircuit]):
+        """Sets the ansatz.
 
-    @VariationalAlgorithm.quantum_instance.setter
+        Args:
+            ansatz: The parameterized circuit used as an ansatz.
+
+        """
+        self._ansatz = ansatz
+        if ansatz is not None:
+            if self._sort_parameters_by_name:
+                self._ansatz_params = sorted(ansatz.parameters, key=lambda p: p.name)
+            else:
+                self._ansatz_params = list(ansatz.parameters)
+
+    @property
+    def gradient(self) -> Optional[Union[GradientBase, Callable]]:
+        """Returns the gradient."""
+        return self._gradient
+
+    @gradient.setter
+    def gradient(self, gradient: Optional[Union[GradientBase, Callable]]):
+        """Sets the gradient."""
+        self._gradient = gradient
+
+    @property
+    def quantum_instance(self) -> Optional[QuantumInstance]:
+        """Returns quantum instance."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
     def quantum_instance(
         self, quantum_instance: Union[QuantumInstance, BaseBackend, Backend]
     ) -> None:
         """set quantum_instance"""
-        super(VQE, self.__class__).quantum_instance.__set__(self, quantum_instance)
-
+        if not isinstance(quantum_instance, QuantumInstance):
+            quantum_instance = QuantumInstance(quantum_instance)
+        self._quantum_instance = quantum_instance
         self._circuit_sampler = CircuitSampler(
-            self._quantum_instance, param_qobj=is_aer_provider(self._quantum_instance.backend)
+            quantum_instance, param_qobj=is_aer_provider(quantum_instance.backend)
         )
 
+    @property
+    def initial_point(self) -> Optional[np.ndarray]:
+        """Returns initial point"""
+        return self._initial_point
+
+    @initial_point.setter
+    def initial_point(self, initial_point: np.ndarray):
+        """Sets initial point"""
+        self._initial_point = initial_point
+
     @property
     def expectation(self) -> ExpectationBase:
         """The expectation value algorithm used to construct the expectation measurement from
         the observable."""
-        return self._expectation
+        if self._input_expectation is not None:
+            return self._input_expectation
+
+        return self._factory_expectation
 
     @expectation.setter
     def expectation(self, exp: ExpectationBase) -> None:
```

Comment:
```suggestion
    def expectation(self, exp: Optional[ExpectationBase]) -> None:
```
I think we allow the user to pass None here to revert back to the factory creating the expectation. It was just not in the typehint - seems it has been like this for a while

Reply:
resolved in 77bb3bc